src/compiler.jl

```diff
@@ -1182,24 +1229,43 @@ end
 const leaked_objs = Base.Dict{Int64, Any}()
 
 if VERSION < v"1.8-"
-function runtime_pfor_augfwd(func, dfunc)::Int64
-	# tape = vec{Any}(numthreads())
-	# pfor threads tape[i] = aug(func, dfunc)
-    
-	tt = Tuple{}
+
+function runtime_pfor_fwd(func, dfunc, width)::Cvoid
+    # pfor threads tape[i] = aug(func, dfunc)
+    @show func, dfunc, width
+    flush(stdout)
+
+    tt = Tuple{}
     # TODO with the MI itself, we should be able to hoist the thunk generation into
     # the original compilation (rather than runtime JIT compiling)
-    forward, adjoint = thunk(func, dfunc, Const, tt, Val(API.DEM_ReverseModePrimal))
+    forward = thunk(func, dfunc, Const, tt, Val(API.DEM_ForwardMode), width)
 
-	tapes = Vector{Ptr{Cvoid}}(undef, Base.Threads.nthreads())
+    function fwd()
+        forward()
+    end
+
+    Base.Threads.threading_run(fwd)
```

Comment:
```suggestion
    Base.Threads.threading_run(forward)
```
?

Reply:
Sure, for here -- though this part is under inspection per 1.8, which I'm going to defer to a later PR.